coordinator/src/main/scala/filodb.coordinator/RowSource.scala

```diff
@@ -79,27 +92,72 @@ trait RowSource extends Actor with StrictLogging {
       if (rows.nonEmpty) {
         val maxSeqId = rows.map(_._1).max
         currentHiSeqNo = Math.max(currentHiSeqNo, maxSeqId)
-        coordinatorActor ! CoordinatorActor.IngestRows(dataset, version, rows.map(_._2), currentHiSeqNo)
+        coordinatorActor ! NodeCoordinatorActor.IngestRows(dataset, version, rows.map(_._2), currentHiSeqNo)
         // Go get more rows
         if (currentHiSeqNo - lastAckedSeqNo < maxUnackedRows) {
           self ! GetMoreRows
         } else {
-          logger.debug(s"Over high water mark: currentHi = $currentHiSeqNo, lastAcked = $lastAckedSeqNo")
+          logger.debug(s" ==> waiting: currentHi = $currentHiSeqNo, lastAcked = $lastAckedSeqNo")
+          context.system.scheduler.scheduleOnce(waitingPeriod, self, CheckCanIngest)
+          context.become(waiting)
         }
       } else {
-        logger.debug(s"Marking isDoneReading as true: HiSeqNo = $currentHiSeqNo, lastAcked = $lastAckedSeqNo")
-        isDoneReading = true
+        logger.debug(s" ==> doneReading: HiSeqNo = $currentHiSeqNo, lastAcked = $lastAckedSeqNo")
+        if (currentHiSeqNo == lastAckedSeqNo) {
+          sendFlush()
+          context.become(doneReading)
+        }
+        else { context.become(doneReading) }
       }
```

Comment:
I'd break these case handlings out to private functions and just have

   case a => funcA(a)
   case b => funcB(b)